backend/src/govsg/services/govsg-callback.service.ts

```diff
@@ -59,7 +71,10 @@ const parseWebhook = async (
       message: 'Received status webhook from WhatsApp',
       action,
     })
-    await parseTemplateMessageWebhook(body as WhatsAppTemplateMessageWebhook)
+    await parseTemplateMessageWebhook(
+      body as WhatsAppTemplateMessageWebhook,
+      clientId
+    )
```

Comment:
Sorry, can't tag on the line again. 

But on line 89: `if ('errors' in body) ...`, am I correct to say that `errors` cannot exist with the other types and thats why we don't have to check it before checking for valid types? Also, why don't we throw an error for this case?

Reply:
I think GitHub doesn't allow you to tag multiple lines? I couldn't do it either iirc.

About `if ('errors' in body)`, we probably don't know much more than the docs [here](https://developers.facebook.com/docs/whatsapp/on-premises/guides/webhooks/#errors).

We don't throw errors because this is the expected way of receiving error notifications from WhatsApp. If we throw an error here, [Meta will resend the request](https://developers.facebook.com/docs/whatsapp/on-premises/guides/webhooks/#retry).

Reply:
I think the lines can't be tagged because it was not changed in this PR. 

To me, the docs suggest that error notifications are a specific type of message and so we won't have a case where `'errors' in body` is true and yet there is other content we need to parse. 

If we don't throw errors because it will result in meta resending the request, then why do we throw an error on line 103? In this case, wouldn't meta just resend this erroneous message indefinitely? 😨 

Also if we receive an error via webhook, how is this handled on the FE though? It currently looks to me like this error is not displayed anywhere and silently happens. (i.e, someone using the FE would never know that this error occured?)

Reply:
There's some subtle difference. `errors` is something Meta can send. Everything else is unexpected.

As for whether we should throw `UnexpectedWebhookError`:

For:

- We don't want them to keep resending
- Be alerted of potential attackers

Against:

- Breaking change from Meta and we want to handle the pending requests before they expire (3 days)

I'm not keen to make the decision now, but I can bring this discussion to Slack if that helps.